File: examples/ingestion-example.py
"""
Example: Loading and applying a PIP identity in Python
"""

from pip_sdk import load_identity, load_preferences, apply_preferences

# Example 1: Basic usage
def example1():
    identity_url = 'https://example.com/identity.json'
    
    try:
        identity = load_identity(identity_url)
        apply_preferences(identity['preferences'])
        print('Identity loaded and applied!')
    except ValueError as e:
        print(f'Failed to load identity: {e}')
        # Use default preferences

# Example 2: Loading only preferences
def example2():
    identity_url = 'https://example.com/identity.json'
    
    try:
        prefs = load_preferences(identity_url)
        
        # Apply theme
        if prefs.get('ui', {}).get('theme'):
            theme = prefs['ui']['theme']
            print(f'Applying theme: {theme}')
        
        # Apply tone
        if prefs.get('interaction', {}).get('tone'):
            tone = prefs['interaction']['tone']
            update_messaging(tone)
        
        # Apply automation
        if prefs.get('automation', {}).get('level'):
            level = prefs['automation']['level']
            configure_automation(level)
    except ValueError as e:
        print(f'Failed to load preferences: {e}')

# Example 3: Flask integration
def example3():
    from flask import Flask, render_template
    from pip_sdk import load_preferences
    
    app = Flask(__name__)
    
    @app.route('/')
    def index():
        identity_url = 'https://example.com/identity.json'
        
        try:
            prefs = load_preferences(identity_url)
            return render_template('index.html', preferences=prefs)
        except ValueError:
            return render_template('index.html', preferences=None)
    
    return app

# Helper functions (examples)
def update_messaging(tone):
    tone_map = {
        'formal': 'Please proceed with your task.',
        'casual': 'Hey! Ready to get started?',
        'friendly': 'Welcome! Feel free to explore.',
        'professional': 'You may proceed with your work.',
        'minimal': 'Welcome.'
    }
    print(tone_map.get(tone, tone_map['friendly']))

def configure_automation(level):
    print(f'Automation level set to: {level}')

# Run example
if __name__ == '__main__':
    example1()

File: examples/ingestion-example.js
/**
 * Example: Loading and applying a PIP identity in JavaScript
 */

import { loadIdentity, applyPreferences } from '@pip/sdk';

// Example 1: Basic usage
async function example1() {
  const identityUrl = 'https://example.com/identity.json';
  
  try {
    const identity = await loadIdentity(identityUrl);
    applyPreferences(identity.preferences);
    console.log('Identity loaded and applied!');
  } catch (error) {
    console.error('Failed to load identity:', error.message);
    // Use default preferences
  }
}

// Example 2: With error handling and caching
async function example2() {
  const identityUrl = 'https://example.com/identity.json';
  const cacheKey = `pip-identity-${identityUrl}`;
  
  // Check cache first
  const cached = localStorage.getItem(cacheKey);
  if (cached) {
    const identity = JSON.parse(cached);
    applyPreferences(identity.preferences);
    return;
  }
  
  // Fetch and cache
  try {
    const identity = await loadIdentity(identityUrl);
    localStorage.setItem(cacheKey, JSON.stringify(identity));
    applyPreferences(identity.preferences);
  } catch (error) {
    console.error('Failed to load identity:', error.message);
  }
}

// Example 3: Applying specific preferences
async function example3() {
  const identityUrl = 'https://example.com/identity.json';
  
  try {
    const identity = await loadIdentity(identityUrl);
    const prefs = identity.preferences;
    
    // Apply theme
    if (prefs.ui?.theme) {
      document.documentElement.setAttribute('data-theme', prefs.ui.theme);
    }
    
    // Apply tone
    if (prefs.interaction?.tone) {
      updateMessaging(prefs.interaction.tone);
    }
    
    // Apply automation
    if (prefs.automation?.level) {
      configureAutomation(prefs.automation.level);
    }
  } catch (error) {
    console.error('Failed to load identity:', error.message);
  }
}

// Helper functions (examples)
function updateMessaging(tone) {
  const toneMap = {
    formal: 'Please proceed with your task.',
    casual: 'Hey! Ready to get started?',
    friendly: 'Welcome! Feel free to explore.',
    professional: 'You may proceed with your work.',
    minimal: 'Welcome.'
  };
  console.log(toneMap[tone] || toneMap.friendly);
}

function configureAutomation(level) {
  console.log(`Automation level set to: ${level}`);
}

// Run example
example1();

